map/map_loader/src/lanelet2_map_loader/lanelet2_differential_loader_module.cpp

```diff
@@ -0,0 +1,155 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "map_loader/lanelet2_differential_loader_module.hpp"
+
+#include "lanelet2_local_projector.hpp"
+#include "map_loader/lanelet2_map_loader_node.hpp"
+
+Lanelet2DifferentialLoaderModule::Lanelet2DifferentialLoaderModule(
+  rclcpp::Node * node, const double & center_line_resolution)
+: logger_(node->get_logger()),
+  clock_(node->get_clock()),
+  center_line_resolution_(center_line_resolution)
+{
+  const auto metadata_adaptor = component_interface_utils::NodeAdaptor(node);
+  metadata_adaptor.init_pub(pub_lanelet_map_meta_data_);
+
+  get_differential_lanelet2_maps_service_ = node->create_service<GetDifferentialLanelet2Map>(
+    "service/get_differential_lanelet_map",
+    std::bind(
+      &Lanelet2DifferentialLoaderModule::onServiceGetDifferentialLanelet2Map, this,
+      std::placeholders::_1, std::placeholders::_2));
+}
+
+bool Lanelet2DifferentialLoaderModule::onServiceGetDifferentialLanelet2Map(
+  GetDifferentialLanelet2Map::Request::SharedPtr req,
+  GetDifferentialLanelet2Map::Response::SharedPtr res)
+{
+  std::vector<std::string> lanelet2_paths;
+  for (const auto & id : req->cell_ids) {
+    auto it = std::find_if(
+      lanelet2_file_metadata_dict_.begin(), lanelet2_file_metadata_dict_.end(),
+      [&id](const auto & file) { return file.second.id == id; });
+    if (it == lanelet2_file_metadata_dict_.end()) {
+      continue;
+    }
+    if (!std::filesystem::exists(it->first)) {
+      continue;
+    }
+    lanelet2_paths.push_back(it->first);
+  }
+  if (lanelet2_paths.empty()) {
+    RCLCPP_ERROR(logger_, "Failed to load differential lanelet2 map");
+    return false;
+  }
+
+  const auto map = differentialLanelet2Load(lanelet2_paths);
+  const auto map_bin_msg =
+    Lanelet2MapLoaderNode::create_map_bin_msg(map, lanelet2_paths[0], rclcpp::Clock().now());
+
+  res->lanelet2_cells = map_bin_msg;
+  res->header.frame_id = "map";
+
+  return true;
+}
+
+lanelet::LaneletMapPtr Lanelet2DifferentialLoaderModule::differentialLanelet2Load(
+  std::vector<std::string> & lanelet2_paths)
+{
+  if (projector_info_.value().projector_type != tier4_map_msgs::msg::MapProjectorInfo::LOCAL) {
+    std::unique_ptr<lanelet::Projector> projector =
+      autoware::geography_utils::get_lanelet2_projector(projector_info_.value());
+
+    lanelet::ErrorMessages errors{};
+    lanelet::io_handlers::MultiOsmParser parser(*projector);
+    lanelet::LaneletMapPtr map = parser.parse(lanelet2_paths, errors);
+
+    if (!errors.empty()) {
+      for (const auto & error : errors) {
+        RCLCPP_ERROR_STREAM(rclcpp::get_logger("map_loader"), error);
+      }
+    }
+
+    return map;
+  } else {
+    const lanelet::projection::LocalProjector projector;
+    lanelet::ErrorMessages errors{};
+    lanelet::io_handlers::MultiOsmParser parser(projector);
```

Comment:
Instead of creating a new parser, I would like to parse them one by one and merge them locally